substitutions.bzl

```diff
@@ -0,0 +1,14 @@
+# Each placeholder is used to stamp versions during the build process, replacing the key with it's
+# value pair. These replacements occur during building of `npm_package`, primarily in `package.json`s.
+NPM_PACKAGE_SUBSTITUTIONS = {
+    # Version of the local package being built, generated via the `--workspace_status_command` flag.
+    "0.0.0-PLACEHOLDER": "{BUILD_SCM_VERSION}",
+    # Versions to declare in release `package.json`s engines field
+    "0.0.0-ENGINES-NODE": "^12.20.0 || ^14.15.0 || >=16.10.0",
```

Comment:
Was hoisting the versions to here so that they are more visible as opposed to being hidden in the filter file?

Reply:
I suspect its better to not add a new top-level file named substitutions.bzl as the substitutions are implementation details of pkg_npm; perhaps a top-level constants.bzl which defines all the constants and then use those constants in `tools/defaults.bzl`. Something like

```
RELEASE_ENGINES_NODE = "^12.20.0 || ^14.15.0 || >=16.10.0"
RELEASE_ENGINES_NPM = "^6.11.0 || ^7.5.6 || >=8.0.0"
RELEASE_ENGINES_YARN = ">= 1.13.0"
```

Some constants could be defined for WORKSPACE as well since there are node versions & hard-coded there.

Reply:
I added this in response to this feedback on the previous PR: https://github.com/angular/angular-cli/pull/22425#discussion_r780445465. The angular components repo has a top-level bazel file with some version substitutions.

I could make it a file with just constants. Wdyt @devversion?

Reply:
Oh. You borrowed the pattern from components. That makes sense. I looked in angular/angular and didn't see any top level .bzl files. Their subs are defined in tools/defaults.bzl https://github.com/angular/angular/blob/master/tools/defaults.bzl#L254

Reply:
I see it now https://github.com/angular/components/blob/master/packages.bzl

Reply:
I like `constants.bzl` as its clear what the file is meant for but that is just bike shedding at that point

Reply:
Renamed to `constants.bzl` and moved substitution logic into `tools/defaults.bzl`.

Reply:
Thank you! yeah I personally like the top-level file as this is something that folks can change without needing a deep understanding of the Bazel setup. Constants sounds good to me 👍